bundles/binding/org.openhab.binding.km200/src/main/java/org/openhab/binding/km200/internal/KM200Binding.java

```diff
@@ -0,0 +1,238 @@
+/**
+ * Copyright (c) 2010-2016 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.km200.internal;
+
+import java.util.Dictionary;
+
+import org.apache.commons.lang.StringUtils;
+import org.openhab.binding.km200.KM200BindingProvider;
+import org.openhab.core.binding.AbstractActiveBinding;
+import org.openhab.core.binding.BindingProvider;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.State;
+import org.osgi.service.cm.ConfigurationException;
+import org.osgi.service.cm.ManagedService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.net.InetAddresses;
+
+/**
+ * The KM200 binding connects to a Buderus Gateway Logamatic web KM50/100/200.
+ *
+ * @author Markus Eckhardt
+ *
+ * @since 1.9.0
+ */
+public class KM200Binding extends AbstractActiveBinding<KM200BindingProvider> implements ManagedService {
+
+    private static final Logger logger = LoggerFactory.getLogger(KM200Binding.class);
+
+    private KM200Device device = null;
+
+    @Override
+    public void bindingChanged(BindingProvider provider, String itemName) {
+        super.bindingChanged(provider, itemName);
+        logger.trace("KN200 bindingChanged: {}", bindingsExist());
+
+        conditionalDeActivate();
+    }
+
+    private void conditionalDeActivate() {
+        logger.trace("KN200 conditional deActivate: {}", bindingsExist());
+
+        if (bindingsExist()) {
+            activate();
+        } else {
+            deactivate();
+        }
+    }
+
+    @Override
+    public void activate() {
+        super.activate();
+
+        logger.debug("KM200 binding has been started.");
+    }
+
+    @Override
+    public void deactivate() {
+        super.deactivate();
+
+        logger.debug("KM200 binding has been stopped.");
+    }
+
+    protected void addBindingProvider(KM200BindingProvider bindingProvider) {
+        super.addBindingProvider(bindingProvider);
+    }
+
+    protected void removeBindingProvider(KM200BindingProvider bindingProvider) {
+        super.removeBindingProvider(bindingProvider);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    @SuppressWarnings("rawtypes")
+    public void updated(Dictionary config) throws ConfigurationException {
+        logger.info("Update KM200 Binding configuration, it takes a minute....");
+
+        if (config == null) {
+            logger.info("Configuration is null");
+            return;
+        } else {
+            if (config.isEmpty()) {
+                throw new RuntimeException("No properties in openhab.cfg set!");
```

Comment:
> is it even possible to have an empty config that is not null?

[This doc](https://osgi.org/javadoc/r4v42/org/osgi/service/cm/ManagedService.html) says the Dictionary can be `null`.  It seems possible that it can be an empty `Dictionary`, but there's no reason I can see to handle that differently than missing required/optional config properties.  When the `Dictionary` passed is `null`, just return from updated with no action or log.

On the subject of config properties, please replace all cases of casting the property value to a String, but instead do `Objects.toString(config.get("X"), null)`, since configuring a binding from the console or via Paper UI can put non-`String` config property values in the `Dictionary`.